src/runtime_src/xrt/device/device.h

```diff
@@ -869,6 +869,12 @@ class device : public xrt_device
     return m_hal->getSysfsPath(subdev, entry);
   }
 
+  hal::operations_result<std::string>
+  getSubdevPath(const std::string& subdev, uint32_t idx)
+  {
+    return m_hal->getSubdevPath(subdev, idx);
+  }
+
```

Comment:
No, this API should not be at this level.  It won't be implementable on windows.
It's fine to have the API in linux/shim.*, but not in common code.